src/App.jsx

```diff
@@ -14,45 +16,106 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+export const App = () => {
+  const [sortedGoods, setSortedGoods] = useState([...goodsFromServer]);
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState(null);
+
+  const sortAlphabetically = () => {
+    if (!isReversed || (isReversed && sortType !== 'alphabetical')) {
+      const orderAbc = isReversed
+        ? [...sortedGoods].sort().reverse()
+        : [...sortedGoods].sort();
+
+      setSortedGoods(orderAbc);
+      setSortType('alphabetical');
+    }
+  };
+
+  const sortByLength = () => {
+    if (!isReversed || (isReversed && sortType !== 'length')) {
+      const orderLength = isReversed
+        ? [...sortedGoods].sort((a, b) => b.length - a.length)
+        : [...sortedGoods].sort((a, b) => a.length - b.length);
+
+      setSortedGoods(orderLength);
+      setSortType('length');
+    }
+  };
```

Comment:
Your reversal condition is needed precisely at the stage of the sorting logic:
```suggestion
  const sortByLength = () => {
    if (!isReversed || (isReversed && sortType !== 'length')) {
      const sortedGoods = [...sortedGoods].sort((a, b) => {
        return isReversed ? b.length - a.length : a.length - b.length;
      });

      setSortedGoods(sortedGoods);
      setSortType('length');
    }
  };
```